R/mlr_callbacks.R

```diff
@@ -0,0 +1,128 @@
+#' @title Early Stopping Callback
+#'
+#' @include CallbackTuning.R
+#' @name mlr3tuning.early_stopping
+#'
+#' @description
+#' This [CallbackTuning] integrates early stopping into the hyperparameter tuning of an XGBoost learner.
```

Comment:
I would formulate this slightly differently: This callback implements early stopping, currently supported learners are: ...

Reply:
Otherwise it makes it seem like it is only intended for xgboost

Reply:
Can we implement this generically by giving learners the property `"early_stopping"` and a standardized slot access via a method  `training_iterations` and then this callback checks whether the learner has the property and if so it calls this method?

Reply:
I used "intergrate" because the early stopping still happens in XGBoost. We do not stop the model. We just extract the best-performing iteration. And yes it only works for XGBoost at the moment. I really like your idea to have a slot that points to the best iteration in the model. We can add this in the next release.